backend/dal/Services/Concrete/UserService.cs

```diff
@@ -107,6 +153,61 @@ public AccessRequest AddAccessRequest(AccessRequest request)
             return request;
         }
 
+        /// <summary>
+        /// Update the access request for the current user.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        public AccessRequest UpdateAccessRequest(AccessRequest request)
+        {
+            if (request == null || request.Agencies == null || request.Roles == null) throw new ArgumentNullException(nameof(request));
+            var userId = this.User.GetUserId();
+            var position = request.User.Position;
+            request.User = this.Context.Users.Find(userId) ??
+                throw new KeyNotFoundException("Your account has not been activated.");
+
+            if (request.UserId != userId) throw new NotAuthorizedException(); // Not allowed to update someone elses request.
+
+            // fetch the existing request from the datasource.
+            var entity = this.Context.AccessRequests
+                .Include(a => a.Agencies)
+                .ThenInclude(a => a.Agency)
+                .Include(a => a.Roles)
+                .ThenInclude(r => r.Role)
+                .Include(a => a.User)
+                .FirstOrDefault(a => a.Id == request.Id) ?? throw new KeyNotFoundException();
+
+            if (entity.User.Position != position)
+            {
+                entity.User.Position = position;
+                this.Context.Entry(entity.User).State = EntityState.Modified;
+            }
+
+            // Remove agencies and roles if required.
+            var removeAgencies = entity.Agencies.Except(request.Agencies, new AccessRequestAgencyAgencyIdComparer());
+            if (removeAgencies.Any()) entity.Agencies.RemoveAll(a => removeAgencies.Any(r => r.AgencyId == a.AgencyId));
+
+            var removeRoles = entity.Roles.Except(request.Roles, new AccessRequestRoleRoleIdComparer());
+            if (removeRoles.Any()) entity.Roles.RemoveAll(a => removeRoles.Any(r => r.RoleId == a.RoleId));
+
+            // Add agencies and roles if required.
+            var addAgencies = request.Agencies.Except(entity.Agencies, new AccessRequestAgencyAgencyIdComparer());
+            addAgencies.ForEach(a => entity.Agencies.Add(a));
+
+            var addRoles = request.Roles.Except(entity.Roles, new AccessRequestRoleRoleIdComparer());
+            addRoles.ForEach(r => entity.Roles.Add(r));
+
+            // Copy values into entity.
+            this.Context.Entry(entity).CurrentValues.SetValues(request);
```

Comment:
The `entity` is being tracked so we need to update it with the untracked `request`.